### DISCUSSION TITLE
Why is the refreservation attribute added to the created file system? The value of the used attribute is the same as that of the refreservation attribute.

### DISCUSSION THREAD

**Discussion by hhb777** [openzfs/zfs#13560] [Q&A] [answered]
I use ZFS version 2.0.7.

When I modify the refreservation value, the value of the used attribute will also be modified, as shown in the following figure.

Please: is this a normal attribute setting?

![image](https://user-images.githubusercontent.com/107523912/173746616-e1915d03-df32-4e48-b32b-7422f5d6ea59.png)

![image](https://user-images.githubusercontent.com/107523912/173749036-177a156e-e269-4f9a-b327-e957b83a5369.png)

**Comment by rincebrain** [comment, marked as answer]
The point of {ref,}reservation is "please make sure this dataset can use at least this amount of space {on the live copy, total}", which logically implies "please treat this dataset as though the USED amount for {the live copy, total} is MAX(actual used value, the reservation property) for accounting for how much space everything but this dataset can use".

So yes, that's expected.

**Reply by hhb777** [reply]
I encountered another problem. The volume was also created using the refreservation parameter. I found that its available attribute value is the usable capacity of the pool. Why is the available attribute value created by the file system our configured capacity rather than the usable capacity of the pool? How can I create a file system so that the available attribute value is the usable capacity of the pool like the volume? 
Thank you very much.

![image](https://user-images.githubusercontent.com/107523912/174021288-ba4adae7-179b-4759-8425-6623e97000f5.png)

**Reply by rincebrain** [reply]
You set a quota of 1G.

Quota is an upper bound.

**Reply by hhb777** [reply]
Yes, I set the quota of the file system to 1G, but after the file system is created, the available attribute value is only 1024M, which is not the remaining available capacity of the pool like the volume.

![image](https://user-images.githubusercontent.com/107523912/174234072-741f02d1-d59b-4725-902d-17fb8ed3ae9c.png)

**Reply by rincebrain** [reply]
Let me be more clear.

Quota is an upper bound on space used. If you set that, your free space will be quota - (usedbydataset + usedbychildren + usedbysnapshots)

**Reply by hhb777** [reply]
Sorry, I didn't make it clear.

For example, I have a 9G pool, and then execute the zfs create -o reservation=1073741824 -o quota=1073741824 pooluuid/filesystemuuid command to create a file system. Through zfs get all, I can see that the value of the available attribute in the file system is 1024M, which is the same as what you said, but I used the same parameters to execute the zfs create -o reservation=1073741824 -V 1073741824 pooluuid/voluuid command to create a volume, Through zfs get all, you can see that the value of the available attribute in this volume is 9G. 

Why is the value of the file system available attribute created by the same parameter different from the value of the volume available attribute created? See the following figure for details of creating file systems and volumes.

Thank you very much.

![image](https://user-images.githubusercontent.com/107523912/174520157-aecf63db-46f5-483c-8410-c0f3162acdac.png)

![image](https://user-images.githubusercontent.com/107523912/174520250-92286285-7dbe-413f-bd1a-58fd8c1667e5.png)

**Reply by rincebrain** [reply]
You made it clear, you're just asking "I made this thing red and this thing blue, why is the second thing not red?"

The volume does not have a quota. reservation, refreservation, and volsize are not a quota. So the property would not be restricted like if it had one.

**Reply by hhb777** [reply]
OK, I think I understand. Thank you very much for your help.
